_posts/2020-06-01-rails-fejlesztes-dockerben.md

```diff
@@ -0,0 +1,233 @@
+
+Láthajuk, hogy létrejöttek oldalak, tesztek, valamint egy kontroller is. Főként ezeket fogjuk használni az IDE tesztelés során. A **zöld háromszög** gomb lenyomásával elindítható a webszerver, ami a [localhost:3000](http://localhost:3000/)-es címen elérhető.
+
+![Imgur](https://i.imgur.com/IJgL6cZ.png)
+
+Ezután próbáljuk ki a többi fontosabb eszköz működését az IDE-ben. A debug, miután beállitottuk, hogy docker exec-el legyen használva, egyből használható.
+
+![Imgur](https://i.imgur.com/Dgln72k.png)
+
+Teszteket is könnyedén futhatunk az IDE GUI-ján keresztül. Ehhez navigáljunk egy testeket tartalmazó fájl-hoz, mondjuk **test/controllers/ruby_mines_controller_test.rb**, itt a **jobb egérgomb** lenyomásával megjelennek a kontextusfüggő lehetőségek. Ezekközül válasszuk a **Run Minitest:**-lehetőséget. A tesztek futásának eredményét is megjeleníti a fejesztőkörnyezet. Emellet van lehetőség a tesztek közül csak egyet futttani, valmint debug módban is elindíthatjuk a teszteket.
+
+![Imgur](https://i.imgur.com/IrfSvfK.png)
+
+A code-completion és az inteligens navigáció is működik, bár ezekhez nem szükséges a nyelvi környezet, az IDE önellóan nyújtja ezeket a szolgáltatásokat. Szinte minden lehetőség elérhető a konténerből, amit a natívan használt verzió támogat. Egy kivételt találtam. A nativ ruby-t használó IDE-ben van lehetőség egyes tesztek futtatása után kilistázni a tesztlfedettséget. Valamint az érintett sorokat is szinezi a környezet, annak függvényében, hogy érintette őket az adott futás. Ez a lehetőség még konténeres használat során nem eléhető, de a fejlesztők már tudnak a [problémáról][1].(A PyCharm alkalmazásban már megodldották, hogy elérhető legyen remote használatával ez a funkcionalitás.)
+
+## Visual Studio Code konfigurálás
+
+Nyissuk meg a projectet VSCode-ban. Első lépésként a **telepitsük a szükséges bővítményeket-öket**. Ezek az alábbiak:
+- Remote - Containers
+- Ruby
+- Ruby Solargraph
+
+Miután ezeket telepitettük **inditsuk újra az IDE-t** és inditsuk el **docker-compose up** parancs kiadásávala a konténereket. Ezután a bal oldalon lévő **Docker logóra** kattinva válasszuk ki **rails_developmet_in_containers_web** konténert, és válasszuk az Attach Visual Studio Code lehetőséget. Ezután telepítsük a konténerbe is a megfelelő bővítményeket. Ezt a **felhő gombra** kattinva egyszerűen megtehetjük. Ezután az **Open Folder** gomb használatával nyissuk meg **/test_app** mappát. A Solargraph müködéséhez extra beállításokat kell elvégeznünk, a források között található [linken][2] ez is megtekintehtő,de ezt itt nem írnám le részletesen. Már csak egy lépés van, hogy elkezdhessük használi a fejlesztő környezetet. A **launch.json** fájl-ban vegyük fel a **Rails server, Listen for rdebug-ide, Rails test** konfigurációkat. A VSCode által javasolt előre definiált lehetőségek teljesen megfelelők. Ezek után már használhatjuk az okos navigálást, code-completion-t, valamit debugolhatjuk és tesztelhetjük az alkalmazásunk.
```

Comment:
Miután ezeket telepítettük, **inditsuk újra az IDE-t** és indítsuk el a **docker-compose up** parancs kiadásával a konténereket. Ezután a bal oldalon lévő **Docker logóra** kattintva válasszuk ki **rails_developmet_in_containers_web** konténert, és válasszuk az Attach Visual Studio Code lehetőséget. Ezután telepítsük a konténerbe is a megfelelő bővítményeket. Ezt a **felhő gombra** kattintva egyszerűen megtehetjük. Ezután az **Open Folder** gomb használatával nyissuk meg **/test_app** mappát. A Solargraph működéséhez extra beállításokat kell elvégeznünk, a források között található [linken][2] ez is megtekinthető, de ezt itt nem írnám le részletesen. Már csak egy lépés van, hogy elkezdhessük használni a fejlesztőkörnyezetet. A **launch.json** fájlban vegyük fel a **Rails server, Listen for rdebug-ide, Rails test** konfigurációkat. A VSCode által javasolt előre definiált lehetőségek teljesen megfelelők. Ezek után már használhatjuk az okos navigálást, code-completion-t, valamit debugolhatjuk és tesztelhetjük az alkalmazásunkat.